homeassistant/components/accuweather/__init__.py

```diff
@@ -0,0 +1,130 @@
+"""The AccuWeather component."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from accuweather import AccuWeather, ApiError, InvalidApiKeyError, RequestsExceededError
+from aiohttp.client_exceptions import ClientConnectorError
+from async_timeout import timeout
+
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_FORECAST,
+    CONF_FORECAST,
+    COORDINATOR,
+    DOMAIN,
+    UNDO_UPDATE_LISTENER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["weather"]
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured AccuWeather."""
+    return True
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})
    return True
```

Reply:
What is the difference?

Reply:
this defines the data domain when integration is setup rather than attempting each time a config entry is setup

Reply:
You're right, it'll be better this way.